### PR TITLE
Fix populating @first and @last in eachAlive helper

### PR DESCRIPTION
**Pull request by reset**
Filter all alive members before populating @first and @last handlebars
fields in eachAlive helper. This fixes an issue where eachAlive would
actually populate @first with a member who is not alive

tenor-255573479

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

File: components/sup/src/templating/helpers/each_alive.rs [hunk 1]
```diff
@@ -35,32 +35,34 @@ impl HelperDef for EachAliveHelper {
             );
             let rendered = match (value.value().is_truthy(), value.value()) {
                 (true, &Json::Array(ref list)) => {
-                    let len = list.len();
+                    let alive_members: Vec<Json> = list.iter()
+                        .filter_map(|m| {
```
Comment (fnichol) [inline components/sup/src/templating/helpers/each_alive.rs]: Love me that `.filter_map`

**Comment by fnichol** [comment]
@thesentinels approve

**Comment by thesentinels** [comment]
:metal: I am testing your branch against master before merging it. We do this to ensure that the master branch is never failing tests.

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started testing this PR](https://travis-ci.org/habitat-sh/habitat/builds/257982370).

**Comment by mwrock** [comment]
oh my...ouch

**Comment by mwrock** [comment]
very nice catch.

@thesentinels approve

**Comment by thesentinels** [comment]
:metal: I am testing your branch against master before merging it. We do this to ensure that the master branch is never failing tests.

**Comment by thesentinels** [comment]
:sparkling_heart: Travis CI [reports this PR passed](https://travis-ci.org/habitat-sh/habitat/builds/257982370).

It always makes me feel nice when humans approve of one anothers work. I'm merging this PR now.

I just want you and the contributor to answer me one question:

gif-keyboard-3280869874741411265

**Pull request merged**